torch/csrc/quantized/quantized_backward.cpp

```diff
@@ -0,0 +1,76 @@
+#include <ATen/native/quantized/PackedParams.h>
+#include <torch/library.h>
+#include <torch/torch.h>
+
+namespace {
+using namespace torch::autograd;
+using namespace at;
+
+class PackedLinearWeightDynamicBackward :
+public Function<PackedLinearWeightDynamicBackward> {
+ public:
+  // Note that both forward and backward are static functions
+
+  // bias is an optional argument
+  static torch::Tensor forward(
+    AutogradContext *ctx,
+    at::Tensor input,
+    const c10::intrusive_ptr<LinearPackedParamsBase>& packed_weight,
+    bool reduce_range) {
+
+    static auto op =
+        at::Dispatcher::singleton()
+            .findSchemaOrThrow("quantized::linear_dynamic", "")
+            .typed<at::Tensor(
+              at::Tensor,
+              c10::intrusive_ptr<LinearPackedParamsBase,
+              c10::detail::intrusive_target_default_null_type<LinearPackedParamsBase>> const&,
+              bool)>();
+    auto output = op.redispatch(
+      DispatchKeySet({DispatchKey::CPU}), input, packed_weight, reduce_range);
+    auto [weight, bias] = packed_weight->unpack();
+    TORCH_CHECK(weight.qscheme() == c10::kPerTensorAffine);
+    // TO-DO : passing a bias does not work, hence create an arbitrary one in backward loop
+    ctx->save_for_backward({weight});
+    return output;
+  }
+
+  static tensor_list backward(AutogradContext* ctx, tensor_list grad_outputs) {
+    auto saved = ctx->get_saved_variables();
+    auto original_weight = saved[0];
+    original_weight = at::permute(original_weight, {1, 0});
+    auto grad_output = grad_outputs[0];
+    static auto op =
```

Comment:
This is just calling a regular function, you can do it via `quantized::linear_prepack(original_weight, bias);`

The forward above is special because it is called while going down the dispatcher. So it needs to nicely redispatch.
This one is called by the autograd engine as regular code. So you can just do regular calls.

Reply:
> The forward above is special because it is called while going down the dispatcher. So it needs to nicely redispatch.

@albanD what is the "nicely redispatch" part?

Reply:
> @albanD what is the "nicely redispatch" part?

Getting the `DispatchKeySet` argument from the caller and calling `redispatch` with the properly updated keyset.

Since you're not using codegen here, you indeed don't have a `quantized::` namespace. Then you can just update the line below to use `call()` instead of `redispatch()`.